lib/user.js

```diff
@@ -55,17 +71,41 @@ const instanceMethods = {
         });
     },
 
-    get functions() {
+    remoteMongoClient(serviceName) {
         const user = this;
-        return new Proxy({}, {
-            get(target, name) {
-                return (...args) => {
-                    return user.callFunction(name, args);
+        return {
+            db(dbName) {
+                return {
+                    collection(collName) {
+                        return new RemoteMongoDBCollection(
+                            user._functionsOnService(serviceName),
+                            dbName,
+                            collName);
+                    },
                 };
             },
+        };
+    },
+
+    _functionsOnService(service) {
```

Comment:
This is basically the same as the old `get functions()` getter but now takes a `service` and uses the property name filtering logic taken from https://github.com/realm/realm-js/blob/11f690217253a095cf605779d006f0be9f7cb6a1/packages/realm-web/src/FunctionsFactory.ts#L149-L158